src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
The original problem here was that calling total was not thread safe. The set, while not being thread safe, is only ever called from one thread (if this ever turns out to be NOT the case, we have another problem. Setting of this should never be done from anything other than the thread that updates the index/aux data).